web_src/js/components/RepoActionView.test.js

```diff
@@ -11,20 +12,29 @@ test('processConsoleLine', () => {
   expect(ansiLogToHTML('\rx\rabc')).toEqual('x\nabc');
   expect(ansiLogToHTML('\rabc\rx\r')).toEqual('abc\nx');
 
-  expect(ansiLogToHTML('\x1b[30mblack\x1b[37mwhite')).toEqual('<span style="color:#000">black<span style="color:#AAA">white</span></span>');
-  expect(ansiLogToHTML('<script>')).toEqual('&lt;script&gt;');
-
+  expect(ansiLogToHTML('\x1b[30mblack\x1b[37mwhite')).toEqual(
+    '<span style="color:rgb(0,0,0)">black</span><span style="color:rgb(255,255,255)">white</span>'
+  );
+  expect(ansiLogToHTML('<script>')).toEqual(
+    '<span style="color:rgb(255,255,255)">&lt;script&gt;</span>'
+  );
 
-  // upstream AnsiToHTML has bugs when processing "\033[1A" and "\033[1B", we fixed these control sequences in our code
-  // if upstream could fix these bugs, we can remove these tests and remove our patch code
-  const ath = new AnsiToHTML({escapeXML: true});
-  expect(ath.toHtml('\x1b[1A\x1b[2Ktest\x1b[1B\x1b[1A\x1b[2K')).toEqual('AtestBA'); // AnsiToHTML bug
-  expect(ath.toHtml('\x1b[1A\x1b[2K\rtest\r\x1b[1B\x1b[1A\x1b[2K')).toEqual('A\rtest\rBA'); // AnsiToHTML bug
+  expect(ansi_up.ansi_to_html('\x1b[1A\x1b[2Ktest\x1b[1B\x1b[1A\x1b[2K')).toEqual(
+    'test'
+  );
+  expect(ansi_up.ansi_to_html('\x1b[1A\x1b[2K\rtest\r\x1b[1B\x1b[1A\x1b[2K')).toEqual(
+    '\rtest\r'
+  );
 
-  // test our patched behavior
-  expect(ansiLogToHTML('\x1b[1A\x1b[2Ktest\x1b[1B\x1b[1A\x1b[2K')).toEqual('test');
-  expect(ansiLogToHTML('\x1b[1A\x1b[2K\rtest\r\x1b[1B\x1b[1A\x1b[2K')).toEqual('test');
+  expect(ansiLogToHTML('\x1b[1A\x1b[2Ktest\x1b[1B\x1b[1A\x1b[2K')).toEqual(
+    '<span style="color:rgb(255,255,255)">test</span>',
+  );
+  expect(ansiLogToHTML('\x1b[1A\x1b[2K\rtest\r\x1b[1B\x1b[1A\x1b[2K')).toEqual(
+    '<span style="color:rgb(255,255,255)">test</span>'
+  );
 
   // treat "\033[0K" and "\033[0J" (Erase display/line) as "\r", then it will be covered to "\n" finally.
-  expect(ansiLogToHTML('a\x1b[Kb\x1b[2Jc')).toEqual('a\nb\nc');
+  expect(ansiLogToHTML('a\x1b[Kb\x1b[2Jc')).toEqual(
+    '<span style="color:rgb(255,255,255)">a</span><span style="color:rgb(255,255,255)">b</span><span style="color:rgb(255,255,255)">c</span>'
```

Comment:
Newlines gone in this test case because I removed `ansiRegexpNewLine`. Wonder whether to restore.

Reply:
It needs to keep the newlines.

Otherwise some progress output are rendered in one line which makes a pretty long line like:

```
Download .... 1% Download .... 10% Download .... 50% Download .... 99% Download .... 100%
```


With the new line, the expected output is:

```
Download .... 1%
Download .... 10%
Download .... 50%
Download .... 99%
Download .... 100%`
```


Context: https://github.com/go-gitea/gitea/pull/23820

Reply:
I will restore it, thanks

Reply:
Restored in https://github.com/go-gitea/gitea/pull/25401/commits/872b9f9fbe8ac917f53942f43a8e794a56a0ed83.